tiledb/sm/c_api/tiledb.cc

```diff
@@ -3864,6 +3877,18 @@ void tiledb_subarray_free(tiledb_subarray_t** subarray) {
   }
 }
 
+int32_t tiledb_subarray_set_layout(
```

Comment:
@stavros, @seth I feel like we had this discussion for query as well. Shouldn't the layout only be set on the query object and now new subarray partitioner object, and not the subarray? Feels like this would make it confusing for the customer?

Reply:
The SubarrayPartitioner 'next()' functionality (called from compute_partition_series()) references the layout of an assoc'd subarray without a query being involved.  Hence, if a user is to be able to request partitions for more than whatever 'default' layout is provided this functionality must be available for the subarray.